### PR TITLE
Python/hf example notebook

### PR DESCRIPTION
**Pull request by awharrison-28**
### Motivation and Context
This PR adds a notebook to the python samples demonstrating how to use SK with hugging face models.

### Description
- add notebook 07-hugging-face-for-skills.ipynb

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: samples/notebooks/python/07-hugging-face-for-skills.ipynb [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  "cells": [
+    {
+      "attachments": {},
+      "cell_type": "markdown",
+      "id": "68e1c158",
+      "metadata": {},
+      "source": [
+        "# Using Hugging Face With Skills\n",
+        "\n",
+        "In this notebook, we demonstrate using hugging face models for Skills using both SemanticMemory and text completions."
```
Comment (alexchaomander) [inline samples/notebooks/python/07-hugging-face-for-skills.ipynb:11]: Capitalize HuggingFace

Comment (alexchaomander) [reply samples/notebooks/python/07-hugging-face-for-skills.ipynb:11]: Also add one more line explaining how this allows you to use local models (downloaded from Hugging Face) so you do not need to use an external API or be connected to the internet.

**Review by alexchaomander — APPROVED** [review]
LGTM. Just add suggested lines to the top of the notebook

File: samples/notebooks/python/07-hugging-face-for-skills.ipynb (hunk 1 above)
Comment (awharrison-28) [reply samples/notebooks/python/07-hugging-face-for-skills.ipynb]: Hopefully my edits are good.

**Review by alexchaomander — APPROVED** [review]
LGTM

**Pull request merged**